src/GdbCommandHandler.cc

```diff
@@ -193,10 +203,21 @@ static string gdb_escape(const string& str) {
 }
 static string gdb_unescape(const string& str) {
   stringstream ss;
-  for (size_t i = 0; i < str.size(); i += 2) {
-    ss << (char)stoul(str.substr(i, 2), nullptr, 16);
+  size_t len = str.size();
+  // care for unexpected string length
+  if (len % 2) {
+    len--;
+    ss << "!";
+  }
+  for (size_t i = 0; i < len; i += 2) {
+    const string hex_char = str.substr(i, 2);
+    try {
+      ss << (char)stoul(hex_char, nullptr, 16);
```

Comment:
Let's use `strtoul` instead of catching C++ exceptions.

Reply:
done